[Bug]: `normalize_tool_call` duplicates an existing `git` command prefix

### Is there an existing issue for the same bug?

- [x] I have searched existing issues and this is not a duplicate.

### Bug Description

In `openhands-sdk/openhands/sdk/agent/utils.py`, `normalize_tool_call` unconditionally prepends `git` when normalizing the `git` tool alias to `terminal`. Given an already complete command such as `{"command": "git status"}`, the function changes the command to `git git status`.

### Expected Behavior

Given `tool_name="git"`, `arguments={"command": "git status"}`, and `available_tools={"terminal"}`, `normalize_tool_call` is expected to return:

```python
("terminal", {"command": "git status"})
```

The existing regression coverage establishes that the subcommand form `tool_name="git"` with `command="status"` should become `git status`. When the command already begins with the exact `git` executable token, normalization should not add a second prefix.


### Actual Behavior

`normalize_tool_call` returns:

```python
("terminal", {"command": "git git status"})
```

The normalized command contains the `git` executable twice. In `git git status`, the second `git` occupies the Git subcommand position instead of `status`.

### Steps to Reproduce

1. Check out Software Agent SDK main at commit `f09e03eac772290feeb51b7d7390ffaefeca1a09`.

2. Add the following test to `tests/sdk/agent/test_tool_call_compatibility.py`:

```python
def test_git_alias_does_not_duplicate_an_existing_command_prefix():
    from openhands.sdk.agent.utils import normalize_tool_call

    tool_name, arguments = normalize_tool_call(
        "git",
        {"command": "git status"},
        {"terminal"},
    )

    assert tool_name == "terminal"
    assert arguments["command"] == "git status"
```

3. Run:

```bash
uv run pytest tests/sdk/agent/test_tool_call_compatibility.py -q
```

4. Observe that the newly added assertion fails while the 26 existing tests pass.

### Installation Method

Source checkout; focused pytest reproduction

### If you selected "Other", please specify

_No response_

### SDK Version

main@f09e03eac772290feeb51b7d7390ffaefeca1a09 (source package version 1.42.0)

### Version Confirmation

- [x] I have confirmed this bug exists on the LATEST version of OpenHands SDK

### Python Version

3.13.2

### Model Name (if applicable)

Not applicable; reproduced by directly invoking deterministic tool-call normalization code

### Operating System

MacOS

### Logs and Error Messages

```text
FAILED tests/sdk/agent/test_tool_call_compatibility.py::test_git_alias_does_not_duplicate_an_existing_command_prefix
AssertionError: assert 'git git status' == 'git status'

- git status
+ git git status
?    ++++

1 failed, 26 passed
```


### Minimal Code Sample

```python
from openhands.sdk.agent.utils import normalize_tool_call

result = normalize_tool_call(
    "git",
    {"command": "git status"},
    {"terminal"},
)

print(result)
# ('terminal', {'command': 'git git status'})
```

### Screenshots and Additional Context

At the tested commit, the terminal-alias branch constructs the normalized command unconditionally:

```python
if original_command:
    normalized_arguments["command"] = f"{base_name} {original_command}"
```

The existing `test_git_alias_executes_terminal_tool` establishes that `tool_name="git"` with `command="status"` should normalize to `git status`. The observed case uses an alternative tool-call shape in which `command` already contains the executable name.

The production agent path calls `normalize_tool_call` while converting a model tool call into an `ActionEvent`. The returned arguments are subsequently used to construct the terminal action, so the duplicated prefix is propagated beyond the normalization helper.

One possible fix direction is to prepend the alias only when the command does not already begin with the same executable as a complete command token. The check should use a token boundary rather than a plain string-prefix comparison, so a value such as `github ...` is not mistaken for an existing `git` prefix. Regression coverage should retain the existing `command="status"` behavior and add the observed `command="git status"` case.

--- a/openhands-sdk/openhands/sdk/agent/utils.py
+++ b/openhands-sdk/openhands/sdk/agent/utils.py
@@ -484,19 +484,6 @@ def _maybe_rewrite_as_terminal_command(
     return tool_name
 
 
-def _starts_with_executable(command: Any, executable: str) -> bool:
-    """Whether ``command``'s first whitespace-delimited token is ``executable``.
-
-    Comparing whole tokens keeps ``github status`` from looking like it already
-    carries a ``git`` prefix.
-    """
-    if not isinstance(command, str):
-        return False
-
-    tokens = command.split(maxsplit=1)
-    return bool(tokens) and tokens[0] == executable
-
-
 def normalize_tool_call(
     tool_name: str,
     arguments: dict[str, Any],
@@ -537,7 +524,9 @@ def normalize_tool_call(
                 }
                 if not original_command:
                     normalized_arguments["command"] = base_name
-                elif _starts_with_executable(original_command, base_name):
+                elif str(original_command).split(maxsplit=1)[:1] == [base_name]:
+                    # Already carries the executable, e.g. git(command="git status").
+                    # Comparing whole tokens keeps "github status" prefixed.
                     normalized_arguments["command"] = original_command
                 else:
                     normalized_arguments["command"] = f"{base_name} {original_command}"

--- a/openhands-sdk/openhands/sdk/agent/utils.py
+++ b/openhands-sdk/openhands/sdk/agent/utils.py
@@ -484,6 +484,19 @@ def _maybe_rewrite_as_terminal_command(
     return tool_name
 
 
+def _starts_with_executable(command: Any, executable: str) -> bool:
+    """Whether ``command``'s first whitespace-delimited token is ``executable``.
+
+    Comparing whole tokens keeps ``github status`` from looking like it already
+    carries a ``git`` prefix.
+    """
+    if not isinstance(command, str):
+        return False
+
+    tokens = command.split(maxsplit=1)
+    return bool(tokens) and tokens[0] == executable
+
+
 def normalize_tool_call(
     tool_name: str,
     arguments: dict[str, Any],
@@ -522,10 +535,12 @@ def normalize_tool_call(
                     for key, value in arguments.items()
                     if key in {"security_risk", "summary"}
                 }
-                if original_command:
-                    normalized_arguments["command"] = f"{base_name} {original_command}"
-                else:
+                if not original_command:
                     normalized_arguments["command"] = base_name
+                elif _starts_with_executable(original_command, base_name):
+                    normalized_arguments["command"] = original_command
+                else:
+                    normalized_arguments["command"] = f"{base_name} {original_command}"
         elif "terminal" in available_tools:
             terminal_command = _maybe_rewrite_as_terminal_command(
                 tool_name,

--- a/tests/sdk/agent/test_tool_call_compatibility.py
+++ b/tests/sdk/agent/test_tool_call_compatibility.py
@@ -704,6 +704,48 @@ def test_reset_alias_executes_terminal_tool(tmp_path):
     assert getattr(action_event.action, "command") == "reset clear"
 
 
+def test_git_alias_keeps_an_already_complete_command(tmp_path):
+    """Regression: a command that already names the executable is left alone."""
+    events = _run_tool_call(
+        tmp_path,
+        tool_name="git",
+        arguments={"command": "git status"},
+        tool_names=(TERMINAL_TOOL_SPEC,),
+    )
+
+    action_event = next(e for e in events if isinstance(e, ActionEvent))
+    errors = [e for e in events if isinstance(e, AgentErrorEvent)]
+
+    assert not errors
+    assert action_event.tool_name == TERMINAL_TOOL_NAME
+    assert action_event.action is not None
+    assert getattr(action_event.action, "command") == "git status"
+
+
+@pytest.mark.parametrize(
+    ("tool_name", "command", "expected"),
+    [
+        ("git", "status", "git status"),
+        ("git", "git status", "git status"),
+        ("git", "git", "git"),
+        ("git", "\tgit\tstatus", "\tgit\tstatus"),
+        ("git", "github status", "git github status"),
+        ("git", "gitk", "git gitk"),
+        ("reset", "clear", "reset clear"),
+        ("reset", "reset clear", "reset clear"),
+    ],
+)
+def test_terminal_prefix_alias_only_adds_a_missing_executable(
+    tool_name, command, expected
+):
+    normalized_tool_name, arguments = agent_utils.normalize_tool_call(
+        tool_name, {"command": command}, {TERMINAL_TOOL_NAME}
+    )
+
+    assert normalized_tool_name == TERMINAL_TOOL_NAME
+    assert arguments["command"] == expected
+
+
 def test_shell_tool_name_git_falls_back_to_terminal(tmp_path):
     """Test that 'git' without arguments falls back to terminal."""
     events = _run_tool_call(

--- a/tests/sdk/agent/test_tool_call_compatibility.py
+++ b/tests/sdk/agent/test_tool_call_compatibility.py
@@ -704,24 +704,6 @@ def test_reset_alias_executes_terminal_tool(tmp_path):
     assert getattr(action_event.action, "command") == "reset clear"
 
 
-def test_git_alias_keeps_an_already_complete_command(tmp_path):
-    """Regression: a command that already names the executable is left alone."""
-    events = _run_tool_call(
-        tmp_path,
-        tool_name="git",
-        arguments={"command": "git status"},
-        tool_names=(TERMINAL_TOOL_SPEC,),
-    )
-
-    action_event = next(e for e in events if isinstance(e, ActionEvent))
-    errors = [e for e in events if isinstance(e, AgentErrorEvent)]
-
-    assert not errors
-    assert action_event.tool_name == TERMINAL_TOOL_NAME
-    assert action_event.action is not None
-    assert getattr(action_event.action, "command") == "git status"
-
-
 @pytest.mark.parametrize(
     ("tool_name", "command", "expected"),
     [

